node/resolvers/search/offer.ts

```diff
@@ -27,10 +43,25 @@ export const resolvers = {
         criteria = InstallmentsCriteria.MAX
       }
 
-      const filteredInstallments = !rates
+      let filteredInstallments = !rates
         ? Installments
         : Installments.filter(({ InterestRate }) => !InterestRate)
 
+      if (includedPaymentSystems) {
+        filteredInstallments = filteredInstallments.filter(
+          ({ PaymentSystemName }) =>
+            includesCaseInsensitive(includedPaymentSystems, PaymentSystemName)
+        )
+      }
+
+      if (excludedPaymentSystems) {
+        filteredInstallments = filteredInstallments.filter(
+          ({ PaymentSystemName }) =>
+            (includesCaseInsensitive(excludedPaymentSystems, PaymentSystemName) === false)
```

Comment:
`=== false` is more visible than a meager `!` at the start of the line

Reply:
Why filter by name, is the ID not available?

Reply:
Yep, the ID is not available in this type.

Also, `PaymentSystemName` seems more intuitive to me. The use case we based this on came from a request like "We want to fetch the installment option with the maximum number of payments, no interest **and that is not Cartão Carrefour**"